sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/SchemaProviderRegistrar.java

```diff
@@ -0,0 +1,42 @@
+package org.apache.beam.sdk.schemas;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.google.auto.service.AutoService;
+import java.util.List;
+import java.util.ServiceLoader;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Experimental.Kind;
+
+/**
+ * {@link SchemaProvider} creators have the ability to automatically have their {@link
+ * SchemaProvider schemaProvider} registered with this SDK by creating a {@link ServiceLoader} entry
+ * and a concrete implementation of this interface.
+ *
+ * <p>It is optional but recommended to use one of the many build time tools such as {@link
+ * AutoService} to generate the necessary META-INF files automatically.
+ */
+@Experimental(Kind.SCHEMAS)
+public interface SchemaProviderRegistrar {
+  /**
+   * Returns a list of {@link SchemaProvider schema providers} which will be registered by default
```

Comment:
I'm curious as to why this always returns the providers that are "registered by default" instead of just the providers that are registered. Why is this extra restriction necessary?

Reply:
I'm not sure I understand. This returns "a" list of providers that will be registered by default. Users will put instances of this in their own modules. e.g. the GCP module can contain one of these to register schemas for all TableRow objects.